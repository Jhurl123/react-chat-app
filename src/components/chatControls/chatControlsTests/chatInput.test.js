import React from 'react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import { mount, configure, shallow } from 'enzyme';
import ChatInput from '../chatInput';
import Adapter from 'enzyme-adapter-react-16';
 
configure({ adapter: new Adapter() });

describe('Chat Input Tests', () => {

  it('renders the UI without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatInput />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ChatInput />)
      .toJSON();
      expect(tree).toMatchSnapshot();  
  })

  it('input is created as expected', () => {
    const wrapper = mount(<ChatInput />);
    const input = wrapper.find('input');
    expect(input).toHaveLength(1);
    expect(input.prop('type')).toEqual('text');
    expect(input.prop('name')).toEqual('message');
  })

  it('submits add message form', () => {

      const userObject = {userId: '123', token: '123123123123123'}
      const fakeEvent = { preventDefault: () => {} };
      const chatComponent = shallow(<ChatInput userObject={userObject}/>);
      const text = chatComponent.find('input')
      text.value = 'test'
      expect(chatComponent.find('form').length).toBe(1);
      chatComponent.find('form').simulate('submit', fakeEvent);

  })
});