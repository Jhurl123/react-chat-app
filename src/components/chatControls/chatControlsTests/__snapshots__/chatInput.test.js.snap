// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Input Tests renders the UI as expected 1`] = `
<div>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "padding": ".9rem",
      }
    }
  >
    <input
      className="makeStyles-ChatInput-1"
      name="message"
      onChange={[Function]}
      placeholder="Send a message"
      type="text"
      value=""
    />
    <button
      aria-label="Send Message"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginLeft": ".5rem",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
      </span>
    </button>
    <button
      aria-label="Attach a File"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginLeft": ".5rem",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
          />
        </svg>
      </span>
    </button>
  </form>
</div>
`;
